src/commands.ts

```diff
@@ -781,4 +784,136 @@ export async function getVersion() {
     }
     
     this.resume();
+}
+class GiftCard {
+    alias: string;
+    cloudId: string;
+    // balance: number = 0;
+    sk: string;
+    address: string;
+    svg: string;
+    url:string;
+
+    constructor(alias: string, cloudId: string, sk: string, address: string, svg: string, url: string) {
+        // this.balance = 0;
+        this.alias = alias
+        this.cloudId = cloudId;
+        this.sk = sk;
+        this.address = address;
+        this.svg = svg;
+        this.url = url;
+    }
+}
+
+export async function generateGiftCards(prefix: string, quantity: string, cardBalance: string, authToken: string) {
+
+    this.pause();
+    const cloudUrl = process.env.CLOUD_API_ENDPOINT;
+    console.log("cloudUrl = ", cloudUrl)
+    const [total] = await this.account.getShieldedBalances(true);
+    const requiredTotalSum = this.account.humanToShielded(cardBalance) * BigInt(quantity);
+
+    if (requiredTotalSum > total) {
+        this.echo(`total card balance ${requiredTotalSum} exceeds available funds ${this.account.shieldedToHuman(total)}`)
+        return
+    }
+
+    const headers = new Headers();
+    headers.append("Authorization", `Bearer ${authToken}`);
+    headers.append("Content-Type", "application/json");
+    let giftCards: GiftCard[] = [];
+    const treeIndex = (await this.account.getPoolTreeState()).index
+    try {
```

Comment:
```suggestion
    try {
        this.echo(`Generating account${Number(quantity) > 1 ? 's' : ''}...`);
```

Progress placeholder